File: test/arguments/without_pubsub_code/cpp/WithoutPubsubCodeTest.cpp
#include <fstream>
#include <string>

#include "gtest/gtest.h"

namespace without_pubsub_code
{

class WithoutPubsubCode : public ::testing::Test
{
protected:
    bool isFilePresent(const char* fileName)
    {
        std::ifstream file(fileName);
        return file.is_open();
    }
};

TEST_F(WithoutPubsubCode, checkService)
{
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Service.cpp"));
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Service.h"));
}

TEST_F(WithoutPubsubCode, checkResponse)
{
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Response.cpp"));
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Response.h"));
}

TEST_F(WithoutPubsubCode, checkRequest)
{
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Request.cpp"));
    ASSERT_TRUE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Request.h"));
}

TEST_F(WithoutPubsubCode, checkPubsub)
{
    ASSERT_FALSE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Pubsub.cpp"));
    ASSERT_FALSE(isFilePresent(
            "arguments/without_pubsub_code/gen/without_pubsub_code/Pubsub.h"));
}

} // namespace without_pubsub_code

File: test/arguments/without_pubsub_code/java/without_pubsub_code/WithoutPubsubCodeTest.java
package without_pubsub_code;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.File;

public class WithoutPubsubCodeTest
{
    @Test
    public void checkService()
    {
        assertTrue(isFilePresent("../gen/without_pubsub_code/Service.java"));
    }

    @Test
    public void checkRequest()
    {
        assertTrue(isFilePresent("../gen/without_pubsub_code/Request.java"));
    }

    @Test
    public void checkResponse()
    {
        assertTrue(isFilePresent("../gen/without_pubsub_code/Response.java"));
    }

    @Test
    public void checkPubsub()
    {
        assertFalse(isFilePresent("../gen/without_pubsub_code/Pubsub.java"));
    }

    private boolean isFilePresent(String fileName)
    {
        File file = new File(fileName);
        return file.exists();
    }
};
